src/classes/hotel.js

```diff
@@ -0,0 +1,110 @@
+import Customer from './customer';
+import Room from './room';
+import Booking from './booking';
+import roomsData from '../../test/test-data/room-data';
+
+class Hotel {
+  constructor(roomsData, customersData, bookingsData) {
+    this.roomsData = roomsData;
+    this.customersData = customersData;
+    this.bookingsData = bookingsData;
+    this.allRooms = [];
+    this.allBookings = [];
+    this.allCustomers = [];
+  }
+  
+  addRooms() {
+    this.roomsData.forEach(roomData => {
+      let room = new Room(roomData);
+      this.allRooms.push(room);
+      return room;
+    })
+  }
+
+  addBookings() {
+    this.bookingsData.forEach(bookingData => {
+      let booking = new Booking(bookingData);
+      this.allBookings.push(booking);
+      return booking;
+    })
+  }
+
+  addCustomers() {
+    this.customersData.forEach(customerData => {
+      let customer = new Customer(customerData);
+      this.allCustomers.push(customer);
+      return customer;
+    })
+  }
+
+  filterBookingsByCustomerID(customer) {
+    let filteredBookingsById = this.allBookings.filter(booking => customer.id === booking.userID);
+    return filteredBookingsById
+  }
+
+  sumUserBookings(customer) {
+    let filteredBookingsById = this.allBookings.filter(booking => customer.id === booking.userID);
+    let sum = filteredBookingsById.reduce((sum, booking) => {
+      this.allRooms.forEach(room => {
+        if (room.roomNumber === booking.roomNumber) {
+          sum += room.cost
+        }
+      })
+      return sum
+    }, 0)
+    return sum
+  }
+
+  filterRoomsByDate(dateInput) {
+    let formatInputDate = dateInput.split('-');
+    let properInputDate = formatInputDate.join('/');
+    let filteredRoomsByDate = []
+    this.allRooms.filter(room => {
+      this.allBookings.forEach(booking => {
+        if(properInputDate === booking.date) {
+          if((booking.roomNumber !== room.roomNumber) && !filteredRoomsByDate.includes(room)) {
+            filteredRoomsByDate.push(room)
+          } else {
+            return filteredRoomsByDate
+          }
+        } else {
+          return filteredRoomsByDate
+        }
+      })
+    })
+    if (filteredRoomsByDate.length !== 0) {
+      return filteredRoomsByDate
+    } else {
+      return `We are profusely sorry.`
+    }
+  }
+
+  filterRoomsByType(roomTypeInput, dateInput) {
```

Comment:
If you have a property on the Hotel (like this.filteredRoomsByDate -very original) that STORES your data from your filterRoomsByDate method THEN you wouldn't have to repeat all this logic here. You can simply iterate over the this.filteredRoomsByDate array to get you what you need.